tests/test_auth.py

```diff
@@ -193,27 +196,55 @@ class TestMustBeContributorDecorator(AuthAppTestCase):
     def setUp(self):
         super(TestMustBeContributorDecorator, self).setUp()
         self.contrib = AuthUserFactory()
-        self.project = ProjectFactory()
-        self.project.add_contributor(self.contrib, auth=Auth(self.project.creator))
-        self.project.save()
+        self.non_contrib = AuthUserFactory()
+        self.public_project = ProjectFactory(is_public=True)
+        self.private_project = ProjectFactory(is_public=False)
+        self.public_project.add_contributor(self.contrib, auth=Auth(self.public_project.creator))
+        self.private_project.add_contributor(self.contrib, auth=Auth(self.private_project.creator))
+        self.public_project.save()
+        self.private_project.save()
+
+    def test_must_be_contributor_when_user_is_contributor_and_public_project(self):
+        result = view_that_needs_contributor(
+            pid=self.public_project._primary_key,
+            user=self.contrib)
+        assert_equal(result, self.public_project)
 
-    def test_must_be_contributor_when_user_is_contributor(self):
+    @unittest.skip('Decorator function bug fails this test, skip until bug is fixed')
+    def test_must_be_contributor_when_user_is_not_contributor_and_public_project_raise_error(self):
+        with assert_raises(HTTPError):
+            view_that_needs_contributor(
+                pid=self.public_project._primary_key,
+                user=self.non_contrib
+            )
+
+    def test_must_be_contributor_when_user_is_contributor_and_private_project(self):
         result = view_that_needs_contributor(
-            pid=self.project._primary_key,
+            pid=self.private_project._primary_key,
             user=self.contrib)
-        assert_equal(result, self.project)
+        assert_equal(result, self.private_project)
 
-    def test_must_be_contributor_when_user_is_not_contributor_raises_error(self):
-        non_contributor = AuthUserFactory()
+    def test_must_be_contributor_when_user_is_not_contributor_and_private_project_raise_error(self):
         with assert_raises(HTTPError):
             view_that_needs_contributor(
-                pid=self.project._primary_key,
-                user=non_contributor
+                pid=self.private_project._primary_key,
+                user=self.non_contrib
             )
 
-    def test_must_be_contributor_no_user(self):
+    def test_must_be_contributor_no_user_and_public_project_redirect(self):
+        res = view_that_needs_contributor(
+            pid=self.public_project._primary_key,
+            user=None,
+        )
+        assert_is_redirect(res)
+        # redirects to login url
+        redirect_url = res.headers['Location']
+        login_url = cas.get_login_url(service_url='http://localhost/')
```

Comment:
...though I see that we're already doing it elsewhere, and it would require parsing out the `:5000` part of `settings.DOMAIN`. This is fine, disregard.